source/common/http/async_client_utility.h

```diff
@@ -39,5 +39,58 @@ class AsyncClientRequestTracker {
   absl::flat_hash_set<AsyncClient::Request*> active_requests_;
 };
 
+/**
+ * Sidestream watermark callback implementation for stream filter that either handles decoding only
+ * or handles both encoding and decoding.
+ */
+class StreamFilterSidestreamWatermarkCallbacks : public Http::SidestreamWatermarkCallbacks {
```

Comment:
Just to highlight: Here is the implementation of SidestreamWatermarkCallbacks.

It can either just push back on downstream via `StreamDecoderFilterCallbacks` or push back on upstream via `StreamEncoderFilterCallbacks` based on client filter configuration (whether handles decoding only or handles both encoding and decoding)